apps/crn-server/test/data-providers/contentful/teams.data-provider.test.ts

```diff
@@ -0,0 +1,589 @@
+import { GraphQLError } from 'graphql';
+import {
+  getContentfulGraphqlClientMockServer,
+  Environment,
+} from '@asap-hub/contentful';
+
+import {
+  getContenfulGraphqlTeam,
+  getContentfulTeamsGraphqlResponse,
+  getTeamCreateDataObject,
+  getTeamDataObject,
+} from '../../fixtures/teams.fixtures';
+import { getContentfulGraphqlClientMock } from '../../mocks/contentful-graphql-client.mock';
+import { getContentfulEnvironmentMock } from '../../mocks/contentful-rest-client.mock';
+import {
+  TeamContentfulDataProvider,
+  teamUnreadyResponse,
+} from '../../../src/data-providers/contentful/teams.data-provider';
+import { getEntry } from '../../fixtures/contentful.fixtures';
+
+describe('Teams data provider', () => {
+  const contentfulGraphqlClientMock = getContentfulGraphqlClientMock();
+  const environmentMock = getContentfulEnvironmentMock();
+  const contentfulRestClientMock: () => Promise<Environment> = () =>
+    Promise.resolve(environmentMock);
+
+  const teamDataProvider = new TeamContentfulDataProvider(
+    contentfulGraphqlClientMock,
+    contentfulRestClientMock,
+  );
+
+  const contentfulGraphqlClientMockServer =
+    getContentfulGraphqlClientMockServer({
+      Teams: () => getContenfulGraphqlTeam(),
+    });
+
+  const teamDataProviderMock = new TeamContentfulDataProvider(
+    contentfulGraphqlClientMockServer,
+    contentfulRestClientMock,
+  );
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  const getContentfulTeamDataObject = () => ({
+    ...getTeamDataObject(),
+    ...teamUnreadyResponse,
+  });
+
+  describe('Fetch method', () => {
+    test('Should fetch the list of teams from Contentful GraphQl', async () => {
+      const result = await teamDataProviderMock.fetch({});
```

Comment:
Why are we testing a mock
```suggestion
      const result = await teamDataProvider.fetch({});
```

Reply:
related to this thread https://github.com/yldio/asap-hub/pull/2759#discussion_r1138422905

Reply:
actually Gabi - Riccardo's suggestion is valid, it's just that this is NOT a mock and I think you got the naming wrong. As you said in another comment in here we have two  TeamContentDataProvider instances:
1. First one is using a graphql simple jest mock
2. Second one is using a mocked graphql server instead which enables us to pass through actual queries and get mocked data back, with graphql query validation

In Squidex tests we used different naming conventions to differentiate them, see:
https://github.com/yldio/asap-hub/blob/260775c669f63fc7b6d0cffbc15c171042e944f6/apps/crn-server/test/data-providers/teams.data-provider.test.ts#L26-L32

Perhaps you might want to rename `teamDataProviderMock` to `teamDataProviderMockGraphql` which indicates that the mocked part is the graphql server?